### PR TITLE
Use transparent inline in the library

### PR DESCRIPTION
**Pull request by nicolasstucki**
The first commit contains the interesting changes. The second only updates the syntax in the tests to make sure that the new version works properly.

### CODE REVIEW

File: library/src/dotty/DottyPredef.scala [hunk 1]
```diff
@@ -8,7 +8,7 @@ object DottyPredef {
       assertFail(message)
   }
 
-  inline final def assert(inline assertion: Boolean) <: Unit = {
+  transparent inline final def assert(inline assertion: Boolean): Unit = {
     if (!assertion)
       assertFail()
   }
```
[the remark this answers is not in the document]
Comment (nicolasstucki) [reply library/src/dotty/DottyPredef.scala:14]: If you write assert(false) the inclined expression becomes of type Nothing.

[the remark this answers is not in the document]
Comment (nicolasstucki) [reply library/src/dotty/DottyPredef.scala:14]: It is like a ??? With a message

[the remark this answers is not in the document]
Comment (nicolasstucki) [reply library/src/dotty/DottyPredef.scala:14]: Not sure if this should be supported thought

File: compiler/src/dotty/tools/dotc/parsing/Parsers.scala [hunk 2]
```diff
@@ -3278,6 +3278,7 @@ object Parsers {
             leadingVparamss ::: rparamss
         var tpt = fromWithinReturnType {
           if in.token == SUBTYPE && mods.is(Inline) && AllowOldWhiteboxSyntax then
+            deprecationWarning("`<:` return type will no longer be suported. Use transparent modifier insead.")
```
Comment (DieBauer) [inline compiler/src/dotty/tools/dotc/parsing/Parsers.scala:3281]:
```suggestion
            deprecationWarning("`<:` return type will no longer be supported. Use transparent modifier instead.")
```

File: compiler/src/dotty/tools/dotc/parsing/Parsers.scala [hunk 3]
```diff
@@ -3547,6 +3548,7 @@ object Parsers {
           mods1 |= Final
           DefDef(name, tparams, vparamss, tpt, subExpr())
         if in.token == USCORE && AllowOldWhiteboxSyntax then
+          deprecationWarning("`<:` return type will no longer be suported. Use transparent modifier insead.")
```
Comment (DieBauer) [inline compiler/src/dotty/tools/dotc/parsing/Parsers.scala:3551]:
```suggestion
            deprecationWarning("`<:` return type will no longer be supported. Use transparent modifier instead.")
```

Comment (nicolasstucki) [reply compiler/src/dotty/tools/dotc/parsing/Parsers.scala]: Thank you

**Pull request merged**